p2p/net/swarm/swarm_listen.go

```diff
@@ -12,13 +13,44 @@ import (
 	ma "github.com/multiformats/go-multiaddr"
 )
 
+type OrderedListener interface {
+	// Transports optionally implement this interface to indicate the relative
+	// ordering that listeners should be setup. Some transports may optionally
+	// make use of other listeners if they are setup. e.g. WebRTC may reuse the
+	// same UDP port as QUIC, but only when QUIC is setup first.
+	// lower values are setup first.
+	ListenOrder() int
+}
+
```

Comment:
Does this matter for webrtc? 
There are two cases here
1. User specifies port: User listens on `/udp/4001/webrtc-direct` and `/udp/4001/quic-v1`
2. User doesn't specify port: User listens on `/udp/0/webrtc-direct` and `/udp/0/quic-v1`

In both the cases quic listener can start after webrtc listener without any impact.